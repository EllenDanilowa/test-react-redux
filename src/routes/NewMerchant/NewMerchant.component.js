import React from 'react';
import PropTypes from 'prop-types';
import MerchantForm from '../../components/MerchantForm/MerchantForm';

const NewMerchant = ({createMerchant}) => {
  const create = (merchant) => createMerchant(merchant);

  return (
    <div>
      <MerchantForm submit={create}
                    title="Create a new merchant"
                    submitTitle="Create"/>
    </div>
  );
};

NewMerchant.propTypes = {
  createMerchant: PropTypes.func.isRequired
};

export default NewMerchant;
