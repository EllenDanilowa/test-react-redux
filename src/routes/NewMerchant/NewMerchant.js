import {connect} from 'react-redux';
import NewMerchant from './NewMerchant.component';
import {createMerchant} from '../../redux/merchant/merchant.actions';

const mapDispatchToProps = (dispatch) => ({
  createMerchant: (merchant) => {
    dispatch(createMerchant(merchant));
  }
});

export default connect(() => ({}), mapDispatchToProps)(NewMerchant);
