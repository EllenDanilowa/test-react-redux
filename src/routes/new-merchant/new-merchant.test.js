import React from 'react';
import renderer from 'react-test-renderer';
import NewMerchant from './new-merchant.component';

jest.mock('../../components/merchant-form/merchant-form', () => () => 'Merchant Form component');

describe('NewMerchant', () => {
  let element;
  let createMerchant;

  const createElement = (props) => {
    const template = (<NewMerchant {...props}/>);

    return renderer.create(template).toJSON();
  };

  beforeEach(() => {
    createMerchant = jest.fn();
  });

  it('renders component correctly', () => {
    element = createElement({createMerchant});

    expect(element).toMatchSnapshot();
  });
});
