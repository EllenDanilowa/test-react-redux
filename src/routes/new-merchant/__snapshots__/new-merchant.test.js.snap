// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMerchant renders component correctly 1`] = `
<div
  className="sc-AxirZ cSToxy"
>
  <h3
    className="sc-AxjAm hnoMwN"
  >
    Create a new merchant
  </h3>
  Merchant Form component
</div>
`;
